src/components/Projects/ProjectsFilterAndSort.tsx

```diff
@@ -0,0 +1,132 @@
+import { t, Trans } from '@lingui/macro'
+import { Collapse, Select } from 'antd'
+import CollapsePanel from 'antd/lib/collapse/CollapsePanel'
+import { CSSProperties, useContext, useState } from 'react'
+
+import { FilterOutlined } from '@ant-design/icons'
+import { ThemeContext } from 'contexts/themeContext'
+
+import FilterCheckboxItem from './FilterCheckboxItem'
+
+type OrderByOption = 'createdAt' | 'totalPaid'
+
+export default function ProjectsFilterAndSort({
+  includeV1,
+  setIncludeV1,
+  includeV1_1,
+  setIncludeV1_1,
+  includeActive,
+  setIncludeActive,
+  includeArchived,
+  setIncludeArchived,
+  orderBy,
+  setOrderBy,
+}: {
+  includeV1: boolean
+  setIncludeV1: Function
+  includeV1_1: boolean
+  setIncludeV1_1: Function
+  includeActive: boolean
+  setIncludeActive: Function
+  includeArchived: boolean
+  setIncludeArchived: Function
+  orderBy: OrderByOption
+  setOrderBy: any
+}) {
+  const {
+    theme: { colors },
+    isDarkMode,
+  } = useContext(ThemeContext)
+
+  const [activeKey, setActiveKey] = useState<0 | undefined>()
+
+  const filterCollapseStyle: CSSProperties = {
+    border: 'none',
+    marginTop: 0,
+    marginBottom: 0,
+    marginRight: 15,
+  }
+
+  // If includeArchived is true, we set active back to true
+  // to avoid both being unselected
+  const toggleArchived = () => {
+    setIncludeActive(includeArchived)
+    setIncludeArchived(!includeArchived)
+  }
+
+  // Close collapse when clicking anywhere in the window except the collapse items
+  window.addEventListener('click', () => setActiveKey(undefined), false)
```

Comment:
This is _somewhat_ fine. Personally, I prefer something like a `useOutsideClick` hook that takes a ref to an element and binds a click event to the window where the click event checks the event's `path`. That way you don't have to worry about click event priority where clicking _within_ the opened menu actually closes it _before_ the menu item's click event. In _this particular_ case, React kinda saves you from this based on its synthetic click event priorities and your use of `stopPropagation`, but consider this unstable.

Another issue I have with this is React will continually bind a new event listener to the window _every time this component renders_. It's not a massive problem, but I consider it a bad practice. This code should look like this instead:

```suggestion
  // Close collapse when clicking anywhere in the window except the collapse items
  useEffect(() => {
    function handleClick() {
      setActiveKey(undefined)
    }

    window.addEventListener('click', handleClick)
    return () => window.removeEventListener('click', handleClick)
  }, [])
```

Reply:
Awesome @torvusbug thanks for putting the change together and giving such a detailed explanation  🙌